crates/iceberg/src/spec/view_metadata_builder.rs

```diff
@@ -0,0 +1,1566 @@
+                Error::new(
+                    ErrorKind::DataInvalid,
+                    "Cannot set current version id to last added version: no version has been added.",
+                )
+            })?;
+        }
+        let version_id = version_id; // make immutable
+
+        if version_id == self.metadata.current_version_id {
+            return Ok(self);
+        }
+
+        let version = self.metadata.versions.get(&version_id).ok_or_else(|| {
+            Error::new(
+                ErrorKind::DataInvalid,
+                format!(
+                    "Cannot set current version to unknown version with id: {}",
+                    version_id
+                ),
+            )
+        })?;
+
+        self.metadata.current_version_id = version_id;
+
+        if self.last_added_version_id == Some(version_id) {
+            self.changes.push(ViewUpdate::SetCurrentViewVersion {
+                view_version_id: Self::LAST_ADDED,
+            });
+        } else {
+            self.changes.push(ViewUpdate::SetCurrentViewVersion {
+                view_version_id: version_id,
+            });
+        }
+
+        self.history_entry = Some(version.log());
+
+        Ok(self)
+    }
+
+    /// Add a new view version and set it as current.
+    pub fn set_current_version(
+        mut self,
+        view_version: ViewVersion,
+        schema: Schema,
+    ) -> Result<Self> {
+        let schema_id = self.add_schema_internal(schema);
+        let view_version = view_version.with_schema_id(schema_id);
+        let view_version_id = self.add_version_internal(view_version)?;
+        self.set_current_version_id(view_version_id)
+    }
+
+    /// Add a new version to the view.
+    ///
+    /// # Errors
+    /// - The schema ID of the version is set to `-1`, but no schema has been added.
+    /// - The schema ID of the specified version is unknown.
+    /// - Multiple queries for the same dialect are added.
+    pub fn add_version(mut self, view_version: ViewVersion) -> Result<Self> {
+        self.add_version_internal(view_version)?;
+
+        Ok(self)
+    }
+
+    fn add_version_internal(&mut self, view_version: ViewVersion) -> Result<i32> {
+        let version_id = self.reuse_or_create_new_view_version_id(&view_version);
+        let view_version = view_version.with_version_id(version_id);
+
+        if self.metadata.versions.contains_key(&version_id) {
+            // ToDo Discuss: Similar to TableMetadata sort-order, Java does not add changes
+            // in this case. I prefer to add changes as the state of the builder is
+            // potentially mutated (`last_added_version_id`), thus we should record the change.
+            if self.last_added_version_id != Some(version_id) {
+                self.changes.push(ViewUpdate::AddViewVersion {
+                    view_version: view_version.with_version_id(version_id),
+                });
+                self.last_added_version_id = Some(version_id);
+            }
```

Comment:
CC @Fokko, @nastra

Reply:
Java returns here:

https://github.com/apache/iceberg/blob/f6faa58dac57e03be6e02a43937ac7c15c770225/core/src/main/java/org/apache/iceberg/view/ViewMetadata.java#L279-L285

So when you do a rollback to an earlier version, the change isn't recorded in Java. Java will only update the `last_added_version_id` if the version is added in the builder. I would assume that the change will be recorded in Java as well. Because I think the `addInBuilder` will only be true if you add it twice. I think I'm okay with this, but let's check with @nastra to be sure 👍

Reply:
what would be the benefit of deviating from what we do on the Java side?